### PR TITLE
Add spankind support

### PR DESCRIPTION
**Pull request by AndrewAXue**
google-cloud-trace has updated to [version 0.24.0](https://pypi.org/project/google-cloud-trace/) and now supports passing in SpanKind information. This PR updates the cloud trace exporter to send this SpanKind info out.

### CODE REVIEW

File: opentelemetry-exporter-google-cloud/src/opentelemetry/exporter/cloud_trace/__init__.py [hunk 1]
```diff
@@ -296,6 +297,21 @@ def _extract_events(events: Sequence[Event]) -> ProtoSpan.TimeEvents:
     )
 
 
+# pylint: disable=no-member
```
Comment (AndrewAXue) [inline opentelemetry-exporter-google-cloud/src/opentelemetry/exporter/cloud_trace/__init__.py:300]: pylint can't find members of Protobuf things

File: opentelemetry-exporter-google-cloud/tests/test_cloud_trace_exporter.py [hunk 2]
```diff
@@ -162,6 +163,8 @@ def test_export(self):
             "time_events": None,
             "start_time": None,
             "end_time": None,
+            # pylint: disable=no-member
```
Comment (AndrewAXue) [inline opentelemetry-exporter-google-cloud/tests/test_cloud_trace_exporter.py:166]: pylint can't find members of Protobuf things

File: opentelemetry-exporter-google-cloud/tests/test_cloud_trace_exporter.py [hunk 3]
```diff
@@ -630,3 +633,24 @@ def test_strip_characters(self):
         self.assertEqual("3.1.0", _strip_characters("3.1.0beta"))
         self.assertEqual("4.2.0", _strip_characters("4b.2rc.0a"))
         self.assertEqual("6.20.15", _strip_characters("b6.20.15"))
+
+    # pylint: disable=no-member
```
Comment (AndrewAXue) [inline opentelemetry-exporter-google-cloud/tests/test_cloud_trace_exporter.py:637]: pylint can't find members of Protobuf things

File: opentelemetry-exporter-google-cloud/src/opentelemetry/exporter/cloud_trace/__init__.py [hunk 4]
```diff
@@ -296,6 +297,21 @@ def _extract_events(events: Sequence[Event]) -> ProtoSpan.TimeEvents:
...
+        return ProtoSpan.SpanKind.INTERNAL
+    if span_kind == trace_api.SpanKind.SERVER:
+        return ProtoSpan.SpanKind.SERVER
+    if span_kind == trace_api.SpanKind.CLIENT:
+        return ProtoSpan.SpanKind.CLIENT
+    if span_kind == trace_api.SpanKind.PRODUCER:
+        return ProtoSpan.SpanKind.PRODUCER
+    if span_kind == trace_api.SpanKind.CONSUMER:
+        return ProtoSpan.SpanKind.CONSUMER
+    return ProtoSpan.SpanKind.SPAN_KIND_UNSPECIFIED
```
Comment (kintel) [inline opentelemetry-exporter-google-cloud/src/opentelemetry/exporter/cloud_trace/__init__.py:312]: Can this be organized as a dict lookup to make it more readable?

**Pull request merged**